syntax = "proto2";

package shardora.view_block.protobuf;

message ViewBlockRequest {
    optional uint32 network_id = 1;
    optional uint32 pool_idx = 2;
}

message ViewBlockItem {
    optional bytes hash = 1;
    optional bytes parent_hash = 2;
    optional uint32 leader_idx = 3;
    optional bytes block_str = 4;
    optional bytes qc_str = 5;
    optional uint32 view = 6;
}

message ViewBlockResponse {
    optional uint32 network_id = 1;
    optional uint32 pool_idx = 2;
    repeated ViewBlockItem view_block_items = 3;
}

message ViewBlockMessage {
    optional ViewBlockRequest view_block_req = 1;
    optional ViewBlockResponse view_block_res = 2;
}
